org.eclipse.wildwebdeveloper.xml/src/org/eclipse/wildwebdeveloper/xml/internal/lsp/ClosingTagReconcilingStrategy.java

```diff
@@ -0,0 +1,115 @@
+package org.eclipse.wildwebdeveloper.xml.internal.lsp;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.DocumentEvent;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.jface.text.ITextViewer;
+import org.eclipse.jface.text.reconciler.DirtyRegion;
+import org.eclipse.jface.text.reconciler.IReconcilingStrategy;
+import org.eclipse.lsp4e.LSPEclipseUtils;
+import org.eclipse.lsp4e.LanguageServiceAccessor;
+import org.eclipse.lsp4e.LanguageServiceAccessor.LSPDocumentInfo;
+import org.eclipse.lsp4j.TextDocumentIdentifier;
+import org.eclipse.lsp4j.TextDocumentPositionParams;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.wildwebdeveloper.xml.internal.XMLLanguageServerAPI;
+
+public class ClosingTagReconcilingStrategy implements IReconcilingStrategy {

```

Comment:
I don't think an IReconcilingStrategy is useful here, why not sticking to a documentListneer (inilined in the IReconciler)?

Reply:
Good catch, I followed your idea with documentListener since the autoclose tag is processed in a CompletableFuture.